### ISSUE TITLE
net/http: close of closed channel

### ISSUE THREAD

**Issue by karmeye** [golang/go#13924]
Since pulling the latest on master branch (109d54a32d15b805769d4c05e78367f126a8d7f0) yesterday I started seeing the below. Did not see them in 1.6 beta 1 distribution.

```
http2: panic serving 127.0.0.1:52421: close of closed channel

runHandler.func1(0xc82d7eff4f, 0xc826580500, 0xc8318685c8)
	/usr/local/go/src/net/http/h2_bundle.go:3737 +0xbc
net/http.(*http2ClientConn).streamByID(0xc8203b2580, 0x100001abb, 0x0)
	/usr/local/go/src/net/http/h2_bundle.go:5043 +0x101
net/http.(*http2ClientConn).forgetStreamID(0xc8203b2580, 0x1abb)
	/usr/local/go/src/net/http/h2_bundle.go:5034 +0x2e
net/http.(*http2ClientConn).RoundTrip(0xc8203b2580, 0xc82d1c7c00, 0xc82930fd48, 0x0, 0x0)
	/usr/local/go/src/net/http/h2_bundle.go:4759 +0x600
net/http.(*http2Transport).RoundTripOpt(0xc8203a2140, 0xc82d1c7c00, 0xc82d104500, 0xc82d104548, 0x0, 0x0)
	/usr/local/go/src/net/http/h2_bundle.go:4441 +0x357
net/http.(*http2Transport).RoundTrip(0xc8203a2140, 0xc82d1c7c00, 0xffffffff, 0x0, 0x0)
	/usr/local/go/src/net/http/h2_bundle.go:4416 +0x41
net/http.http2noDialH2RoundTripper.RoundTrip(0xc8203a2140, 0xc82d1c7c00, 0xc82930fd40, 0x0, 0x0)
	/usr/local/go/src/net/http/h2_bundle.go:244 +0x3b
net/http.(*Transport).RoundTrip(0xc8200f3ce0, 0xc82d1c7c00, 0xc8200f3ce0, 0x0, 0x0)
	/usr/local/go/src/net/http/transport.go:252 +0x352
net/http.send(0xc82d1c7b20, 0xb44500, 0xc8200f3ce0, 0xece27375e, 0x1dcb0d0a, 0x706220, 0xc82930fdd0, 0x0, 0x0)
	/usr/local/go/src/net/http/client.go:260 +0x6b7
net/http.(*Client).send(0xc82000f800, 0xc82d1c7b20, 0xece27375e, 0x1dcb0d0a, 0x706220, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/client.go:155 +0x185
net/http.(*Client).doFollowingRedirects(0xc82000f800, 0xc82d1c7b20, 0x5892f8, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/client.go:474 +0x877
net/http.(*Client).Do(0xc82000f800, 0xc82d1c7b20, 0xc800000002, 0x0, 0x0)
	/usr/local/go/src/net/http/client.go:188 +0xff
```

Probably 1000 - 1500 requests were processing, then the above was logged for each. But the program did not exit.

I am unsure if this is an issue, i.e. if this is an internal error or caused by calling code.

**Comment by bradfitz** [comment]
Thanks for the report! I see how this regressed. The fix is trivial. Adding a test is a bit more work.

**Comment by karmeye** [comment]
Great, thanks!

**Comment by bradfitz** [comment]
https://go-review.googlesource.com/18571

**Comment by bradfitz** [comment]
(still need to bundle into std)

**Comment by mqliang** [comment]
May I ask does Go1.6 has fixed this bug? We use Go1.5 and meet this bug also.

**Comment by ianlancetaylor** [comment]
Yes, this is fixed in Go 1.6.

In general, please do not use the issue tracker to ask questions.  See https://golang.org/wiki/Questions .  Thanks.
